vmm/src/memory_manager.rs

```diff
@@ -228,8 +241,27 @@ impl MemoryManager {
             mem_end.unchecked_add(1)
         };
 
+        let mut virtiomem_region = None;
+        let mut virtiomem_resize = None;
         if let Some(size) = hotplug_size {
-            start_of_device_area = start_of_device_area.unchecked_add(size);
+            if hotplug_method == HotplugMethod::VirtioMem {
+                let start_addr = GuestAddress(
+                    (start_of_device_area.0 + vm_virtio::VIRTIO_MEM_DEFAULT_BLOCK_SIZE - 1)
+                        / vm_virtio::VIRTIO_MEM_DEFAULT_BLOCK_SIZE
+                        * vm_virtio::VIRTIO_MEM_DEFAULT_BLOCK_SIZE,
```

Comment:
Could you use the bitwise operation here?

Reply:
Updated.